devtools/pre-push-hook.sh

```diff
@@ -1,11 +1,13 @@
-#!/bin/sh
+#!/bin/sh -e
 
-# This script checks that the version in the code matches the latest version
-# tag.
-#
+# This script performs some checks.
 # Install as a pre-push hook from the repo root with:
 #   cp devtools/pre-push-hook.sh .git/hooks/pre-push
 
+go test -short ./...
```

Comment:
This is increasingly becoming similar to the GitHub action that we have, which also fires for PRs - so it's more general. 

If you feel this functionality is also needed in the pre-push hook, it may make sense to factor it out into a script that both run? Doesn't have to necessarily happen now since this is still very simple; just something to keep in mind

Reply:
Will look into it. Do you know why the github action does `go build` before `go test`?